inspirehep/modules/orcid/utils.py

```diff
@@ -143,3 +148,32 @@ def get_push_access_token(orcid):
     token, _ = remote_token.token()
 
     return token
+
+
+def get_orcids_for_push(record):
+    """Obtain the ORCIDs associated to the list of authors in the Literature record.
+
+    The ORCIDs are looked up both in the ``ids`` of the ``authors`` and in the
+    Author records that have claimed the paper.
+
+    Args:
+        record(dict): metadata from a Literature record
+
+    Returns:
+        Iterator[str]: all ORCIDs associated to these authors
+    """
+    orcids_on_record = []
+    author_recids_with_claims = []
+
+    for author in record.get('authors', []):
+        orcids_in_author = get_values_for_schema(author.get('ids', []), 'ORCID')
+        if orcids_in_author:
+            orcids_on_record.extend(orcids_in_author)
+        elif author.get('curated_relation') is True and 'record' in author:
+            author_recids_with_claims.append(get_recid_from_ref(author['record']))
+
+    author_records = get_db_records('aut', author_recids_with_claims)
+    all_ids = (author.get('ids', []) for author in author_records)
+    orcids_in_authors = chain.from_iterable(get_values_for_schema(ids, 'ORCID') for ids in all_ids)
+
```

Comment:
You are using 4 utils coming from 3 different repos in a few lines. It was very expensive and annoying for me to follow this code. I know it is not your fault, but IMO utils modules are an anti-pattern for this reason, code should fit in better abstraction.